nhaystack-rt/src/nhaystack/util/TypeUtil.java

```diff
@@ -226,6 +228,46 @@ else if (def instanceof BSimple)
 
             return simple;
         }
+        else if (def instanceof BEnumOverride)
+        {
+            BEnumOverride cpx = (BEnumOverride) def;
+            BFacets facets = comp.getAction(action.getName()).getFacets();
+            BEnumRange range = (BEnumRange) facets.get(BFacets.RANGE);
+
+            Iterator<Map.Entry<String, HVal>> it = args.iterator();
+            while (it.hasNext())
+            {
+                Map.Entry<String, HVal> e = it.next();
+                BSimple value = toBajaSimple(e.getValue());
+                if (!e.getKey().equals("value"))
+                {
+                    cpx.set(e.getKey(), value);
+                    continue;
+                }
+                // construction of BDynamicEnum arg
+                if (value instanceof BString && range.isTag(value.toString()))
+                {
+                    value = range.get(value.toString());

```

Comment:
Referring to line 235 above and also my comment, if 'range' is not set on line 235 to a valid BEnumRange, is it null here at line 250, could this possibly throw an exception?

Reply:
Yes, it could